diff -u libpcap-1.1.1/gencode.c libpcap-1.1.1.tmp/gencode.c
--- libpcap-1.1.1/gencode.c	2010-03-11 19:56:54.000000000 -0600
+++ libpcap-1.1.1.tmp/gencode.c	2015-01-12 15:17:48.283017369 -0600
@@ -139,7 +138,6 @@
 		(void)vsnprintf(pcap_geterr(bpf_pcap), PCAP_ERRBUF_SIZE,
 		    fmt, ap);
 	va_end(ap);
-	longjmp(top_ctx, 1);
 	/* NOTREACHED */
 }

@@ -425,17 +423,6 @@
 	root = NULL;
 	bpf_pcap = p;
 	init_regs();
-	if (setjmp(top_ctx)) {
-#ifdef INET6
-		if (ai != NULL) {
-			freeaddrinfo(ai);
-			ai = NULL;
-		}
-#endif
-		lex_cleanup();
-		freechunks();
-		return (-1);
-	}

 	netmask = mask;

